### PR TITLE
PROD-522: Add credits to mystery box flow

### PR DESCRIPTION
**Pull request by samott**
Updates mystery box flow to include credits,

PR has PROD-553 already merged.

### CODE REVIEW

**Comment by yash431garg** [comment]
@samott can you merge #767 again I have a fix there few code changes?

**Review by neel-ds — DISMISSED** [review]

**Comment by yash431garg** [comment]
@samott I have fixed the changes you requested.

File: app/actions/mysteryBox/open.ts [hunk 1]
```diff
@@ -92,39 +87,40 @@ export async function openMysteryBox(
...
-        throw new Error(
-          `Don't know how to send prize for mystery box ${mysteryBoxId}, type ${prize.prizeType} token ${prize.tokenAddress}`,
-        );
-
       const prizeAmount = Number(prize.amount ?? 0);
-
       let sendTx: string | null;
-
-      if (prizeAmount > 0) {
+      // if (
```
Comment (marvinmarnold) [inline app/actions/mysteryBox/open.ts:115]: Delete unused code

File: app/queries/home.ts [hunk 2]
```diff
@@ -373,3 +374,25 @@ async function queryUsersTotalClaimedAmount(userId: string): Promise<number> {
 
   return Number(result[0].totalClaimedAmount);
 }
+
+/**
+ * Retrieves the total credit amount claimed by the user from mystery box prizes.
+ *
+ * @returns {Promise<number>} The total credit amount claimed by the user.
+ */
+export async function getUsersTotalCreditAmount() {
```
Comment (marvinmarnold) [inline app/queries/home.ts:383]: This is correct if you only want to know how much has been claimed via boxes. But if this is intended to be used as a global counter, then it should look directly at FungibleAssetToken instead

File: prisma/schema.prisma [hunk 3]
```diff
@@ -300,6 +300,7 @@ enum EBoxTriggerType {
   ClaimAllCompleted
   DailyDeckCompleted
   ChompmasStreakAttained
+  TutorialCompleted
```
Comment (marvinmarnold) [inline prisma/schema.prisma:303]: Where in the code do we make sure that each user can only have one trigger with this type?

File: components/MysteryBox/MysteryBox.tsx [hunk 4]
```diff
@@ -141,16 +140,17 @@ function MysteryBox({
...
   };
 
   const handleSkip = async () => {
     if (isSubmitting) return;
 
-    try {
-      if (mysteryBoxId && skipAction == "Dismiss")
-        await dismissMysteryBox(mysteryBoxId);
-    } catch {}
+    if (mysteryBoxId && skipAction == "Dismiss")
```
[the remark this answers is not in the document]
Comment (samott) [reply components/MysteryBox/MysteryBox.tsx:152]: I can change this, but it's a union type, FYI.

File: prisma/schema.prisma (hunk 3 above)
Comment (yash431garg) [reply prisma/schema.prisma:306]: https://github.com/gator-labs/chomp/blob/0f58e3364f2eb0252140ca42706bee0289ef9a90/app/queries/mysteryBox.ts#L66

File: components/MysteryBox/MysteryBox.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (yash431garg) [reply components/MysteryBox/MysteryBox.tsx:152]: What do we mean union type @samott?

**Pull request merged**